ddl/ddl_api.go

```diff
@@ -798,6 +798,10 @@ func (d *ddl) AddColumn(ctx context.Context, ti ast.Ident, spec *ast.AlterTableS
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if col.DefaultValue == nil && mysql.HasNotNullFlag(col.Flag) {
+		col.DefaultValue = table.GetZeroValue(col.ToInfo())
+	}
+	col.OriginDefaultValue = col.DefaultValue
```

Comment:
Should make sure originDefaultValue is not a time function.
OriginDefaultValue should be either string or nil.